### DISCUSSION TITLE
[Web API] How to get the exact size of an element including decimals, regardless of scale?

### DISCUSSION THREAD

**Discussion by MTtankkeo** [community/community#135732] [Programming Help] [answered]
### Body

I'm facing a challenge in accurately measuring the intrinsic size of an HTML element, including decimal places, regardless of any scaling applied. Here's the situation:

1. `clientWidth` and `clientHeight` return the intrinsic size of an element, ignoring any applied scale. However, these methods round to the nearest integer, losing decimal precision.

2. `getBoundingClientRect()` provides dimensions with decimal precision, but it returns the rendered size, which is affected by scaling.

For example, if an element's intrinsic size is 100.5 pixels and a scale of 2 is applied:
- `clientWidth`/`clientHeight` would return 100
- `getBoundingClientRect()` would return 201 (100.5 * 2)

Even if we try to calculate the scale by comparing these values, we can't get an accurate result because `clientWidth`/`clientHeight` have already discarded the decimal portion of the intrinsic size.

Is there any way to retrieve the exact intrinsic size of an element, including decimal places, regardless of any scaling applied?

I've considered various approaches, but haven't found a reliable solution that works across different scaling methods (CSS transforms, zoom, etc.) and provides the level of precision I need.

Any insights or alternative approaches would be greatly appreciated. Thanks.

### Guidelines

- [X] I have read and understood this category's [guidelines](https://github.com/orgs/community/discussions/51384) before making this post.

**Comment by erencanbas** [comment, marked as answer]
**Handling Transformations**

If there are transformations applied (like scaling), you can combine the above method with getBoundingClientRect() to factor out the scale and get the unscaled dimensions.

```
function getIntrinsicSize(element) {
    const rect = element.getBoundingClientRect();
    const computedStyle = getComputedStyle(element);
    const scaleX = rect.width / parseFloat(computedStyle.width);
    const scaleY = rect.height / parseFloat(computedStyle.height);

    const intrinsicWidth = rect.width / scaleX;
    const intrinsicHeight = rect.height / scaleY;

    return { width: intrinsicWidth, height: intrinsicHeight };
}
```

**Reply by MTtankkeo** [reply]
Thanks!, does CSSStyleDeclaration.width return the layout measured intrinsic size? I didn't know haha. You are my lifesaver. 😊
